src/Drivers/SlackNotificationDriver.php

```diff
@@ -97,28 +101,11 @@ public static function isSetup(): bool
      */
     public static function getPrivateChannel() : ?string
     {
-        return SlackUser::find(auth()->user()->group->id)->channel_id;
-    }
+        $slack_user = SlackUser::find(auth()->user()->group_id);
 
-    /**
-     * Return driver_id of public subscription.
-     *
-     * @param string $notification
-     *
-     * @return string
-     */
-    public static function getPublicDriverId(string $notification) : ?string
-    {
-        return null;
-    }
+        if (is_null($slack_user))
+            return null;
 
-    /**
-     * Return the route key which have to be used in a private notification registration flow.
-     *
-     * @return string
-     */
-    public static function getPrivateRegistrationRoute(): ?string
```

Comment:
this method is needed if a user has not setup slack before:
https://github.com/herpaderpaldent/seat-notifications/blob/2.0.x/src/Http/Controllers/SeatNotificationsController.php#L67 

by adding this you need to update https://github.com/herpaderpaldent/seat-notifications/blob/2.0.x/src/Http/Controllers/Slack/SlackUserOAuthController.php. Check the Discord Controller in regard to usage of the session data etc.

i see you have implemented a validation error for this. However the method should be there nontheless and be enforced by the interface/abstract-class

Reply:
It is not "removed".
It's inherited from parent class since the current one isn't implementing it

https://github.com/herpaderpaldent/seat-notifications/pull/55/files/78eeb0bf116c526fe56d5c0a8e3f26de975ce058#diff-0c8339868be97c335225bdf439dc256eR46